python-frontend/typeshed_serializer/serializer/symbols.py

```diff
@@ -61,6 +63,7 @@ class TypeDescriptor:
     def __init__(self, _type: mpt.Type):
         self.args = []
         self.fully_qualified_name = None
+        self.kind = None
```

Comment:
If you had to add this, I assume that means that the final `else` on line 132 can actually be triggered. As this shows, it wasn't a robust piece of code, but back when I wrote this I actually assumed this condition could not be triggered because all "common" type classes encountered in TypeShed were covered (and I just wanted to avoid a failure if it actually wasn't the case due to some unsupported edge case). 

Do you know on which case it happens? Maybe the comment on line 133 can be updated to mention when it is triggered, or maybe there are other type classes we can actually handle here.

**EDIT:** At least one case where it seems to happen is when setting an alias for an overloaded symbol. For example in `gettext.pyi` I see multiple overloaded definitions for the function `translation`, then a variable definition:

```
Catalog = translation
```

This ultimately means we serialize variable `tranlsation` to have a type annotation such as:
```
  type_annotation {
    pretty_printed_name: "#Unknown"
  }
```

Maybe we shouldn't do that and just not serialize any type annotation? Or maybe from the actual type class we see from mypy, we can infer this is an overloaded function and actually serialize that (I see the type class `Overload` actually exist in mypy).

Reply:
Good point! I added `self.kind = None` because indeed I had trouble with variables pointing to aliases of overloaded symbols and I then forgot to handle the "#Unknown" types.
I think we can just avoid to serialize any type annotation in that case, and maybe improve the logic and try to get the actual type class in another ticket / PR.

Reply:
Could you create a ticket for that? I don't think it major but I guess it would provide a small precision boost in some cases.

Reply:
Sure, I created [SONARPY-960](https://jira.sonarsource.com/browse/SONARPY-960)